tests/TestCase/ORM/TableTest.php

```diff
@@ -2778,6 +2778,43 @@ public function testUpdateNoPrimaryButOtherKeys()
     }
 
     /**
+     * Test saveMany() with entities array
+     *
+     * @return void
+     */
+    public function testSaveManyArray()
+    {
+        $entities = [
+            new Entity(['name' => 'admad']),
+            new Entity(['name' => 'dakota'])
+        ];
+
+        $table = TableRegistry::get('authors');
+        $result = $table->saveMany($entities);
+        $this->assertSame($entities, $result);
+        $this->assertTrue(isset($result[0]->id));
+    }
+
+    /**
+     * Test saveMany() with ResultSet instance
+     *
+     * @return void
+     */
+    public function testSaveManyResultSet()
+    {
+        $table = TableRegistry::get('authors');
+
+        $entities = $table->find()->all();
+        $entities->first()->name = 'admad';
+
+        $result = $table->saveMany($entities);
+        $this->assertSame($entities, $result);
+
+        $first = $table->find()->first();
+        $this->assertSame('admad', $first->name);
+    }
```

Comment:
Tried that. The state of 1st entity does get changed. `isNew()` returns false for it even though transaction was rolled back and entity wasn't actually persisted. So what should be done?